src/components/select/Select.tsx

```diff
@@ -26,18 +27,24 @@ import ValidationText from '../validationText';
 import theme from './theme.css';
 import { Option as OptionType } from './types';
 
-const minHeightBySizeMap = {
+const minHeightBySizeMap: Record<string, number> = {
   tiny: 24,
   small: 30,
   medium: 36,
   large: 48,
 };
 
-/* *
- * We have to define BaseSelectProps with optional options, so we can then set them required in Select and never in AsyncSelect.
- * Unfortunately because of BoxProps having index of unknown, Omiting of these props causes all props to become unknown, we cannot Omit options in AsyncSelect.
- * */
-export interface BaseSelectProps<
+export interface SelectProps<
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  Option extends OptionType = OptionType,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  IsMulti extends boolean = false,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  IsClearable extends boolean = false,
+> {
+  [key: string]: any;
+}
+export interface SelectPropsFull<
```

Comment:
Q: when does this stop being temporary? I also don't think `SelectPropsFull` is being used now

Reply:
Ah right I read the description 😄